tests/unittests/dataset_tests.py

```diff
@@ -1028,6 +1029,20 @@ def test_field_schemas(self):
                 "spam", embedded_doc_type=fo.Detections
             )
 
+    @drop_datasets
+    def test_validate_sample_fields(self):
+        dataset = foz.load_zoo_dataset("quickstart")
+        dataset._doc.sample_fields[-1] = None
+        with self.assertRaises(ValidationError):
+            dataset.save()
+
+    @drop_datasets
+    def test_validate_frame_fields(self):
+        dataset = foz.load_zoo_dataset("quickstart-video")
```

Comment:
ditto: use this to avoid downloading a zoo dataset:

```py
dataset = fo.Dataset()
dataset.media_type = "video"
```

Reply:
Done ✅